From 6caad5bafe2c8b8badb90fd61f96ecd871a507fc Mon Sep 17 00:00:00 2001
From: Chaitanya Manapragada <cman39@student.monash.edu.au>
Date: Fri, 2 Jun 2017 23:09:01 +1000
Subject: [PATCH 07/27] Here is proof that alternate nodes are voting.

The -999 check isn't strong enough. It's only good for one level.
---
 src/moa/classifiers/trees/HATADWIN.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/moa/classifiers/trees/HATADWIN.java b/src/moa/classifiers/trees/HATADWIN.java
index 12099e7..179d9ec 100644
--- a/src/moa/classifiers/trees/HATADWIN.java
+++ b/src/moa/classifiers/trees/HATADWIN.java
@@ -658,6 +658,12 @@ public class HATADWIN extends HoeffdingTree {
                 	// So the filter will still add any nodes found deeper down to foundNodes
                 	// This looks like a bug.
 
+                	if (foundNode.node != null){
+                		if (((NewNode)foundNode.node).isAlternate()){
+                			System.err.println("Alternate is being used for prediction");
+                			System.exit(1);
+                		}
+                	}
 
                     Node leafNode = foundNode.node;
                     if (leafNode == null) {
-- 
2.7.4

